src/openassetio-python/private/include/openassetio/private/python/exceptions.hpp

```diff
@@ -115,9 +109,11 @@ struct CppExceptionsAndPyClassNames {
  * Python exception.
  */
 template <class CppException>
-struct HybridException : py::error_already_set, CppException {
+struct PYBIND11_EXPORT HybridException : CppException {
```

Comment:
[question] This is just beyond the edge of my knowledge. The `HybridException` is only ever thrown and caught internally within the `_openassetio` dll, so does the template class _really_ need exporting?  

The wrinkle is that an intermediate dll may intercept the `HybridException`, i.e. catch (as the base type) and re-`throw`, to be caught again in `_openassetio` (as a `HybridException`). I have assumed, without much evidence, that this case requires the explicit symbol export.

The practical reason for adding the export here is that the tests fail (on MacOS / libc++).  This is due to the separate `_openassetio_test` Python extension module reusing `HybridException`, which is a special case that will not occur in production, but has some similarities to the case above, i.e. interception by an intermediate dll.

I think a better test would be to properly simulate the intermediate dll `catch`-and-re`throw` case, but would require some time to implement.  Worth adding?

Reply:
I'm actually struggling to think of how to "slot in" an arbitrary 3rd-party dll that catches and re-throws the exception.  However, it is possible (even likely) that we'll do this in `libopenassetio-core` as part of logging/auditing.  Below are a couple of sequence diagrams to hopefully illustrate:

Current Python<->Python e2e exception propagation
```mermaid
sequenceDiagram
    host.py->>_openassetio (host): Manager.resolve(...)
    _openassetio (host)->>libopenassetio-core: Manager::resolve(...)
    libopenassetio-core->>_openassetio (manager): ManagerInterface::resolve(...)
    _openassetio (manager)->>manager.py: MangerInterface.resolve(...)
    manager.py-->>_openassetio (manager): throw OpenAssetIOException(...)
    _openassetio (manager)-->>libopenassetio-core: throw HybridException(...)
    libopenassetio-core-->>_openassetio (host): propagate exception
    _openassetio (host)-->>host.py: unpack OpenAssetIOException
```

Potential future addition of exception intercept for intermediate logging and re-throw
```mermaid
sequenceDiagram
    host.py->>_openassetio (host): Manager.resolve(...)
    _openassetio (host)->>libopenassetio-core: Manager::resolve(...)
    libopenassetio-core->>_openassetio (manager): ManagerInterface::resolve(...)
    _openassetio (manager)->>manager.py: MangerInterface.resolve(...)
    manager.py-->>_openassetio (manager): throw OpenAssetIOException(...)
    _openassetio (manager)-->>libopenassetio-core: throw HybridException(...)
    libopenassetio-core->>libopenassetio-core: catch OpenAssetIOException + log + re-throw
    libopenassetio-core-->>_openassetio (host): propagate exception
    _openassetio (host)-->>host.py: unpack OpenAssetIOException
```

Reply:
As testing the below flow indicates things work as expected, ie, the only reason we need to export is because of our unrepresentative test module, I think the best thing to do from this is to leave the export, (with a big comment explaining the situation,) and [plan] to address our python test infrastructure so we can remove it.

Reply:
Added issue:

- https://github.com/OpenAssetIO/OpenAssetIO/issues/1237

Reply:
Added an `@todo` with a link to the above issue in b9cbaa9e

Reply:
Perfect